app/src/main/java/agency/five/codebase/android/movieapp/ui/home/HomeScreen.kt

```diff
@@ -0,0 +1,300 @@
+            popularViewState = homeScreenMapper.toHomeMovieCategoryViewState(
+                popularCategoryLabels,
+                MovieCategory.getByOrdinal(it)!!,
+                MoviesMock.getMoviesList()
+            )
+        },
+        onNowPlayingLabelClick = {
+            nowPlayingViewState = homeScreenMapper.toHomeMovieCategoryViewState(
+                nowPlayingCategoryLabels,
+                MovieCategory.getByOrdinal(it)!!,
+                MoviesMock.getMoviesList()
+            )
+        },
+        onUpcomingLabelClick = {
+            upcomingViewState = homeScreenMapper.toHomeMovieCategoryViewState(
+                upcomingCategoryLabels,
+                MovieCategory.getByOrdinal(it)!!,
+                MoviesMock.getMoviesList()
+            )
+        },
+        onNavigateToMovieDetails = onNavigateToMovieDetails
+    )
+}
+
+@Composable
+fun HomeScreen(
+    popularViewState: HomeMovieCategoryViewState,
+    nowPlayingViewState: HomeMovieCategoryViewState,
+    upcomingViewState: HomeMovieCategoryViewState,
+    onPopularLabelClick: (Int) -> Unit,
+    onNowPlayingLabelClick: (Int) -> Unit,
+    onUpcomingLabelClick: (Int) -> Unit,
+    onNavigateToMovieDetails: (Int) -> Unit,
+) {
+    LazyColumn(
+        verticalArrangement = Arrangement.spacedBy(dimensionResource(id = R.dimen.home_screen_spacing))
+    ) {
+        item {
+            PopularMovies(popularViewState, onPopularLabelClick, onNavigateToMovieDetails)
+        }
+
+        item {
+            NowPlayingMovies(nowPlayingViewState, onNowPlayingLabelClick, onNavigateToMovieDetails)
+        }
+
+        item {
+            UpcomingMovies(upcomingViewState, onUpcomingLabelClick, onNavigateToMovieDetails)
+        }
+    }
+}
+
+@Composable
+fun UpcomingMovies(
+    upcomingViewState: HomeMovieCategoryViewState,
+    onLabelClick: (Int) -> Unit,
+    onNavigateToMovieDetails: (Int) -> Unit
+) {
+    Text(
+        text = stringResource(id = R.string.upcoming),
+        fontSize = 17.sp,
+        fontWeight = FontWeight.Bold,
+        color = MaterialTheme.colors.primary,
+        modifier = Modifier
+            .padding(MaterialTheme.spacing.medium)
+    )
+    LazyRow(
+        horizontalArrangement = Arrangement.spacedBy(MaterialTheme.spacing.large),
+        userScrollEnabled = false,
+        modifier = Modifier
+            .padding(dimensionResource(id = R.dimen.medium_spacing))
+    ) {
+        items(upcomingViewState.movieCategories.size) { category ->
+            MovieCategoryLabel(
+                movieCategoryLabelViewState = MovieCategoryLabelViewState(
+                    upcomingViewState.movieCategories[category].itemId,
+                    upcomingViewState.movieCategories[category].isSelected,
+                    upcomingViewState.movieCategories[category].categoryText
+                ),
+                onClick = { onLabelClick(upcomingViewState.movieCategories[category].itemId) },
+                modifier = Modifier
```

Comment:
MovieCategoryLabel should have default parameter Modifier if you don't need to append anything